sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3404,7 +3405,7 @@ piEnqueueKernelLaunch(pi_queue Queue, pi_kernel Kernel, pi_uint32 WorkDim,
   // Save the kernel in the event, so that when the event is signalled
   // the code can do a piKernelRelease on this kernel.
   (*Event)->CommandData = (void *)Kernel;
-
+  Queue->RefCount++;
```

Comment:
This code already set the lock at line 3386, so I can't call piQueueRetain() which internally will try to acquire the lock again.

Reply:
I created a helper function createEventAndAssociateQueue to outline the whole common code section.